src/core/ViewabilityTracker.ts

```diff
@@ -317,18 +354,43 @@ export default class ViewabilityTracker {
 
     //TODO:Talha optimize this
     private _diffUpdateOriginalIndexesAndRaiseEvents(newVisibleItems: number[], newEngagedItems: number[]): void {
-        this._diffArraysAndCallFunc(newVisibleItems, this._visibleIndexes, this.onVisibleRowsChanged);
+        const minimumViewTime = this._viewabilityConfig && this._viewabilityConfig.minimumViewTime
+        ? this._viewabilityConfig.minimumViewTime
+        : 0;
+        this._diffArraysAndCallFunc(newVisibleItems, this._visibleIndexes, this.onVisibleRowsChanged, minimumViewTime);
         this._diffArraysAndCallFunc(newEngagedItems, this._engagedIndexes, this.onEngagedRowsChanged);
         this._visibleIndexes = newVisibleItems;
         this._engagedIndexes = newEngagedItems;
     }
 
-    private _diffArraysAndCallFunc(newItems: number[], oldItems: number[], func: TOnItemStatusChanged | null): void {
+    private checkMinimumViewTime = (all: number[], now: number[], notNow: number[], minimumViewTime: number, callbackFunc: TOnItemStatusChanged): void => {
+        const that = this;
+        const timeoutId = setTimeout(() => {
+            that.timers.delete(timeoutId);
+
+            const currAll = all.filter((index) => that._visibleIndexes.indexOf(index) >= 0);
+            const currNow = currAll.filter((index) => that._lastReportedVisibleIndexes.indexOf(index) === -1);
+            const currNotNow = that._lastReportedVisibleIndexes.filter((index) => currAll.indexOf(index) === -1);
+
+            if (currAll.length > 0 && (currNow.length > 0 || currNotNow.length > 0)) {
+                that._lastReportedVisibleIndexes = currAll;
+                callbackFunc(currAll, currNow, currNotNow);
+            }
+        }, minimumViewTime);
+        this.timers.add(timeoutId);
+    }
+
+    private _diffArraysAndCallFunc(newItems: number[], oldItems: number[], func: TOnItemStatusChanged | null, minimumViewTime?: number): void {
         if (func) {
             const now = this._calculateArrayDiff(newItems, oldItems);
             const notNow = this._calculateArrayDiff(oldItems, newItems);
             if (now.length > 0 || notNow.length > 0) {
-                func([...newItems], now, notNow);
+                // Adding default minimum view time of 250ms for performance optimization
+                if (minimumViewTime && minimumViewTime >= Constants.DEFAULT_MIN_VIEW_TIME) {
```

Comment:
instead of not executing  the minimumViewTime logic incase of DEFAULT_MIN_VIEW_TIME greater than minimumViewTime you could execute with minimumViewTime logic with default value.